### ISSUE TITLE
Bug: [await-thenable] Don't flag values of an unconstrained type parameter 

### ISSUE THREAD

**Issue by JoshuaKGoldberg** [typescript-eslint/typescript-eslint#10311]
### Before You File a Bug Report Please Confirm You Have Done The Following...

- [X] I have tried restarting my IDE and the issue persists.
- [X] I have updated to the latest version of the packages.
- [X] I have [searched for related issues](https://github.com/typescript-eslint/typescript-eslint/issues?q=is%3Aissue+label%3A%22package%3A+eslint-plugin%22) and found none that matched my issue.
- [X] I have [read the FAQ](https://typescript-eslint.io/linting/troubleshooting) and my problem is not listed.

### Playground Link

https://typescript-eslint.io/play/#ts=5.6.2&fileType=.ts&code=IYZwngdgxgBAZgV2gFwJYHsIwO4CdgAOBAprgDwAqAfABQBuwANgsQFwwUCUMA3gFAwYuYsgS4swbMFTIYDZsQDcfAL5A&eslintrc=N4KABGBEBOCuA2BTAzpAXGUEKQAIBcBPABxQGNoBLY-AWhXkoDt8B6AQwHd3K78ALRE3YAjJOiiJo0APbRI4MAF8QSoA&tsconfig=N4KABGBEDGD2C2AHAlgGwKYCcDyiAuysAdgM6QBcYoEEkJemy0eAcgK6qoDCAFutAGsylBm3TgwAXxCSgA&tokens=false

### Repro Code

```TypeScript
async function wrapper<T>(value: T) {
  return await value;
}
```

### ESLint Config

```javascript
module.exports = {
  "rules": {
    "@typescript-eslint/await-thenable": "error"
  }
}
```

### tsconfig

```jsonc
{
  "compilerOptions": {
    "strictNullChecks": true
  }
}
```

### Expected Result

The `await value` shouldn't be reported on. We don't know that `T` is or isn't a Promise/Thenable. I'd like to be able to use the `await` just in case it is.

### Actual Result

On the `await value`:

```plaintext
Unexpected `await` of a non-Promise (non-"Thenable") value.
```

### Additional Info

For a slightly more realistic reproduction, consider a function that `await`s the result of calling a function for a generic value:

```ts
type GetSomething<T> = (timestamp: number) => T;

async function wrapper<T>(getter: GetSomething<T>) {
  const timestamp = performance.now();

  // I'd like to await in here to preserve the call stack
  return await getter(timestamp);
}
```

Somewhat related: #8301, where it was noted that the rule intentionally ignores `any` and `unknown` values because there's no guarantee they aren't a Promise.

💖

**Comment by kirkwaiblinger** [comment]
Just want to add that the same fix will need to be propagated to `return-await` ([playground](https://typescript-eslint.io/play/#ts=5.6.2&fileType=.ts&code=IYZwngdgxgBAZgV2gFwJYHsIwO4CdgAOBAprgDwAqAfABQBuwANgsQFwwUCUMA3gFAwYuYsgS4swbMFTIYDZsQDcfAL5A&eslintrc=N4KABGBEBOCuA2BTAzpAXGUEIHodQAEAXATwAcUBjaASzKIFoV4aA7InAQwHdObGiAC0StOAIyToovaK0gAacNkjFyVWvSbIW7HNERFYshjz5EpkGXKUBfEDaA&tsconfig=N4KABGBEDGD2C2AHAlgGwKYCcDyiAuysAdgM6QBcYoEEkJemy0eAcgK6qoDCAFutAGsylBm3TgwAXxCSgA&tokens=false)), which contains pesky duplicate logic that we periodically threaten to deprecate (for example #9903)
